frontend/src/react-query/api/schema.tsx

```diff
@@ -0,0 +1,189 @@
+
+      return data;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaModeQuery = () => {
+  return useQuery<string | null>({
+    queryKey: ['schemaRegistry', 'mode'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/mode`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema mode');
+      }
+
+      const data = await response.json();
+      if (data.isConfigured === false) {
+        return null;
+      }
+
+      return data.mode ?? null;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaCompatibilityQuery = () => {
+  return useQuery<string | null>({
+    queryKey: ['schemaRegistry', 'compatibility'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/config`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema compatibility');
+      }
+
+      const data = await response.json();
+      if (data.isConfigured === false) {
+        return null;
+      }
+
+      return data.compatibility ?? null;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaDetailsQuery = (subjectName: string, options?: { enabled?: boolean }) => {
+  return useQuery<SchemaRegistrySubjectDetails>({
+    queryKey: ['schemaRegistry', 'subjects', subjectName, 'details'],
+    queryFn: async () => {
+      const response = await fetch(
+        `${config.restBasePath}/schema-registry/subjects/${encodeURIComponent(subjectName)}/versions/all`,
+        {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${config.jwt}`,
+          },
+        },
+      );
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch schema details for ${subjectName}`);
+      }
+
+      return response.json();
+    },
+    staleTime: 30000, // Cache for 30 seconds
+    refetchOnMount: true,
+    enabled: options?.enabled !== false,
+  });
+};
+
+export const useUpdateGlobalCompatibilityMutation = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation<SchemaRegistryConfigResponse, Error, SchemaRegistryCompatibilityMode>({
+    mutationFn: async (mode: SchemaRegistryCompatibilityMode) => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/config`, {
+        method: 'PUT',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ compatibility: mode }),
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to update global compatibility mode');
```

Comment:
Can we use `ConnectError.from(error)` and then formatErrorMessage with gRPC? I know it's RESTful API this time but we can still format our errors in the same way, I believe it would work